src/json-patch-duplex.d.ts

```diff
@@ -5,25 +5,71 @@
  * MIT license
  */
 declare module jsonpatch {
-    function unobserve(root: any, observer: any): void;
-    function observe(obj: any, callback: any): any;
-    function generate(observer: any): any;
+    type Patch = AddPatch | RemovePatch | ReplacePatch | MovePatch | CopyPatch | TestPatch;
+    interface PatchBase {
+        path: string;
+    }
+    interface AddPatch extends PatchBase {
+        op: 'add';
+        value: any;
+    }
+    interface RemovePatch extends PatchBase {
+        op: 'remove';
+    }
+    interface ReplacePatch extends PatchBase {
+        op: 'replace';
+        value: any;
+    }
+    interface MovePatch extends PatchBase {
+        op: 'move';
+        from: string;
+    }
+    interface CopyPatch extends PatchBase {
+        op: 'copy';
+        from: string;
+    }
+    interface TestPatch extends PatchBase {
+        op: 'test';
+        value: any;
+    }
+    type JsonPatchErrorName = 'SEQUENCE_NOT_AN_ARRAY' | 'OPERATION_NOT_AN_OBJECT' | 'OPERATION_OP_INVALID' | 'OPERATION_PATH_INVALID' | 'OPERATION_FROM_REQUIRED' | 'OPERATION_VALUE_REQUIRED' | 'OPERATION_VALUE_CANNOT_CONTAIN_UNDEFINED' | 'OPERATION_PATH_CANNOT_ADD' | 'OPERATION_PATH_UNRESOLVABLE' | 'OPERATION_FROM_UNRESOLVABLE' | 'OPERATION_PATH_ILLEGAL_ARRAY_INDEX' | 'OPERATION_VALUE_OUT_OF_BOUNDS';
```

Comment:
that's the typescript compiler's doing... I don't believe there's anything I can do about that. I could update the d.ts but like you said before, it'll just get overwritten.